utils_linux.go

```diff
@@ -217,8 +218,9 @@ type runner struct {
 }
 
 func (r *runner) run(config *specs.Process) (_ int, retErr error) {
+	detach := r.detach || (r.action == CT_ACT_CREATE)
 	defer func() {
-		if retErr != nil {
+		if !detach || retErr != nil {
```

Comment:
Can you comment why is this change?

Reply:
This is because `run`, for a non-detached container, actually destroys it once it's done. @lifubang just moved this code from the end of `run` to here.

Reply:
Thanks!